ffi/src/lib.rs

```diff
@@ -238,7 +238,7 @@ fn get_from_root(
     key: &Value,
 ) -> Result<Value, String> {
     let db = db.ok_or("db should be non-null")?;
-    let requested_root = root.as_slice().try_into()?;
+    let requested_root = HashKey::try_from(root.as_slice()).map_err(|e| e.to_string())?;
```

Comment:
fit: this looks a bit weird now but it'll look better in a future PR that handles results a bit better